File: loan-repayment-java/src/main/java/com/loan/repayment/service/IRepaymentPlanService.java
package com.loan.repayment.service;

import com.loan.repayment.entity.RepaymentPlanRequest;
import com.loan.repayment.entity.RepaymentPlanResponse;

/**
 *
 * @author Ramachandra
 *
 */
public interface IRepaymentPlanService {

    /**
     * This method calculates the repayment plan.
     *
     * @param repaymentPlanRequest Payload for the repayment loan
     * @return repaymentPlanResponse
     */
    RepaymentPlanResponse repaymentPlan(RepaymentPlanRequest repaymentPlanRequest);
}

File: loan-repayment-kotlin/src/main/kotlin/com/loan/repayment/service/IRepaymentPlanService.kt
package com.loan.repayment.service

import com.loan.repayment.entity.RepaymentPlanRequest
import com.loan.repayment.entity.RepaymentPlanResponse

/**
 *
 * @author Rama chandra
 */
interface IRepaymentPlanService {
    /**
     * This method calculates the repayment plan.
     *
     * @param repaymentPlanRequest Payload for the repayment loan
     * @return repaymentPlanResponse
     */
    fun repaymentPlan(repaymentPlanRequest : RepaymentPlanRequest) : RepaymentPlanResponse
}